src/view/actions.ts

```diff
@@ -5,33 +5,53 @@ import { GitHobsSettings } from 'settings';
 import { Notice, RequestUrlResponse, TFile } from 'obsidian';
 import { GitHubIssueStatus } from 'view';
 
+function formatUnsafeFileName(fileName: string) {
+	return fileName.replace(':', ' - ').replace('|', ' -- ').replace('/', '_').replace('\\', '~');
+}
+
 async function updateFile(
 	file: MarkdownFile,
 	res: RequestUrlResponse,
 	externalData?: string,
 	title?: string
 ) {
 	try {
-		const propertiesWithGithubIssue = PropertiesHelper.writeIssueId(
-			externalData ?? file.data,
+		const filePropertiesString = PropertiesHelper.extractFilePropertiesString(file.data);
+		let propertiesWithGithubIssueData = PropertiesHelper.writeIssueId(
```

Comment:
Since this variable does not have only the _issue id property_, I suggest renaming it to simply `properties`